jobs/integr8ly/ocp3/nightly/osd-testing-nightly-trigger.yaml

```diff
@@ -0,0 +1,206 @@
+                                [$class: 'AmazonWebServicesCredentialsBinding', credentialsId: "${CREDENTIALS_ID}", accessKeyVariable: 'AWS_ACCESS_KEY_ID', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'],
+                                [$class: 'StringBinding', credentialsId: "rhmi-qe1-sso-client-secret", variable: 'SSO_SECRET'],
+                                [$class: 'StringBinding', credentialsId: "rhmi-qe1-token", variable: 'OPENSHIFT_TOKEN']
+                            ]) {
+                            installationStatus = build(job: 'osd-cluster-integreatly-install', propagate: false, parameters: [
+                                string(name: 'clusterName', value: "${CLUSTER_NAME}"),
+                                string(name: 'installationGitBranch', value: "${INSTALLATION_BRANCH}"),
+                                string(name: 'routerShard', value: "${ROUTER_SHARD}"),
+                                string(name: 'openShiftToken', value: "${OPENSHIFT_TOKEN}"),
+                                string(name: 'githubClientId', value: "${GH_CLIENT_ID}"),
+                                string(name: 'githubClientSecret', value: "${GH_CLIENT_SECRET}"),
+                                string(name: 'customerAdminPassword', value: "${CUSTOMER_ADMIN_PASSWORD}"),
+                                string(name: 'threeScaleAwsAccessKey', value: "${AWS_ACCESS_KEY_ID}"),
+                                string(name: 'threeScaleAwsSecretKey', value: "${AWS_SECRET_ACCESS_KEY}"),
+                                string(name: 'threeScaleBucketName', value: "${THREESCALE_BUCKET_NAME}"),
+                                string(name: 'threeScaleBucketRegion', value: "${THREESCALE_BUCKET_REGION}"),
+                                string(name: 'backupAwsKey', value: "${AWS_ACCESS_KEY_ID}"),
+                                string(name: 'backupAwsSecret', value: "${AWS_SECRET_ACCESS_KEY}"),
+                                string(name: 'backupsBucketName', value: "${BACKUPS_BUCKET_NAME}"),
+                                string(name: 'clientSecret', value: "${SSO_SECRET}")
+                            ]).result
+                        }
+                    } // stage
+
+                    if (installationStatus != 'SUCCESS' || uninstallationStatus != 'SUCCESS') {
+                        println 'There were some errors in previous install/uninstall pipelines!'
+                        currentBuild.result = 'FAILURE'
+                        return
+                    }
+
+                    // This relies on a cronjob (or something else) to update the client secret in master-config.yaml so that we can successfully login afterwards
+                    stage ('Try to login to installed cluster') {
+                        println "Waiting for successful login to the ${CLUSTER_NAME} cluster"
+                        i = 0
+                        // wait max 1 hour
+                        while(!loggedIn && i < 360) {
```

Comment:
I would change the comment saying that this is for a specific number of tries (it is not really timeboxed). Eg `// Trying to log into the cluster (twelve tries should take approx. 1 hour)`. Also, if your aim is to have this waiting for roughly one hour, you need to decrease the number of tries to 12.

Reply:
For loop might be more appropriate here.

Reply:
Hmmm I can update the comment but I don't like using break in for loops TBH.